### PR TITLE
Make indexing public

### PR DESCRIPTION
**Pull request by schmidt-sebastian**

### CODE REVIEW

File: firebase-common/src/main/java/com/google/firebase/annotations/PreviewApi.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import java.lang.annotation.Target;
+
+/**
+ * Indicates that this object (class, method, etc) is experimental and that both its signature and
+ * implementation are subject to change. An API marked with this annotation provides no guarantee
+ * of API stability or backward-compatibility.
+ */
+@KeepForSdk
+@Target({ElementType.TYPE, ElementType.FIELD, ElementType.METHOD, ElementType.CONSTRUCTOR})
+public @interface PreviewApi {}
```
Comment (vkryachko) [inline firebase-common/src/main/java/com/google/firebase/annotations/PreviewApi.java:29]: We have a `firebase-annotations` sdk, would it make sense to move it there?

Comment (schmidt-sebastian) [reply firebase-common/src/main/java/com/google/firebase/annotations/PreviewApi.java:29]: I can do this, but I would have to add a dependency on `com.google.android.gms:play-services-basement` to use `@KeepForSdk`.

Comment (vkryachko) [reply firebase-common/src/main/java/com/google/firebase/annotations/PreviewApi.java:29]: Why do you need `@KeepForSdk`? it has no effect in github

Comment (schmidt-sebastian) [reply firebase-common/src/main/java/com/google/firebase/annotations/PreviewApi.java]: Okay, removed "KeepForSdk" and moved the class to firebase-annotations. PTAL.

**Review by 9996142202 — APPROVED** [review]

**Pull request merged**